controller/src/main/java/io/pravega/controller/server/SegmentHelper.java

```diff
@@ -108,7 +109,7 @@ public SegmentHelper(final ConnectionFactory clientCF, HostControllerStore hostS
         final long requestId = connection.getFlow().asLong();
         Pair<Byte, Integer> extracted = extractFromPolicy(policy);
 
-        Map<Class, Runnable> expectedReplies = new HashMap<>();
+        expectedReplies.clear();
```

Comment:
This is not what I asked.

I asked you to create a static, final, read-only HashMap which you create once (statically or in a constructor) and where you pre-populate all these expected responses.

The code you just wrote is not thread safe and it doesn't improve the performance over the previous iteration.

Reply:
thanks Andrei,
• each request expects different different reply i.e., `createSegment` expects `SegmentAlreadyExistes` or `SegmentCreated` but not `SegmentSealed`, if put all these expected responses in a constructor it will be a problem when `SegmentSealed` returns to `createSegment` - as a special case.
• the RawClient instance will be create in each function call, I think a static final HashMap can not take the one time creation benefit.
please correct me.

Reply:
a member Map to hold all the expected replies has been added for avoiding repeat adding expected replies into `expectedReplies` map at every function call, and the related handling also updated. thanks for the suggestion.